### PR TITLE
IGNITE-14607: Regex Based IP Filtering

### PR DESCRIPTION
**Pull request by atris**
This commit introduces the notion of regexes which can be registered and used as filters for incoming IPAddresses. A new API is introduced for the same and implemented in Azure Blob Storage IP Finder.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/TcpDiscoveryIpFinderAdapter.java [hunk 1]
```diff
@@ -71,6 +79,24 @@
...
+        if (regexList == null) {
+            return true;
+        }
+
+        for (Pattern pattern : regexList) {
+            Matcher matcher = pattern.matcher(ipAddress);
+
+            if (matcher.matches()) {
+                return false;
+            }
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/TcpDiscoveryIpFinderAdapter.java:94]:
```suggestion
            if (matcher.matches())
                return false;
```

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/TcpDiscoveryIpFinderAdapter.java [hunk 2]
```diff
@@ -71,6 +79,24 @@
         return shared;
     }
 
+    /** {@inheritDoc} */
+    @Override public boolean isIpAddressAllowed(String ipAddress) {
+
+        if (regexList == null) {
+            return true;
+        }
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/TcpDiscoveryIpFinderAdapter.java:87]:
```suggestion
        if (regexList == null)
            return true;
```

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/TcpDiscoveryIpFinderAdapter.java [hunk 3]
```diff
@@ -85,6 +111,21 @@ public TcpDiscoveryIpFinderAdapter setShared(boolean shared) {
         return this;
     }
 
+    /**
+     * Adds a regex for filtering IPs
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/TcpDiscoveryIpFinderAdapter.java:115]:
```suggestion
     * Adds a regex for filtering IPs.
     *
     * @param regex Regular expression.
     * @return {@code this} for chaining.
```

**Review by AMashenkov — CHANGES_REQUESTED** [review]

**Comment by atris** [comment]
Superseded by https://github.com/apache/ignite/pull/9048

**Pull request closed without merging**